sensors/simsensor_common/simsensor_generic.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Generic functions for all sensor simulators
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "simsensor_generic.h"
+
+#include "event_queue.h"
+#include "simsensor_reader.h"
+
+#include <stdlib.h>
+#include <errno.h>
+#include <time.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/time.h>
+#include <sys/threads.h>
+
+
+typedef struct {
+	char *name;
```

Comment:
If we wish to use string literals here its necessary to make this `char` constant. As underlying memory will be rdonly
```suggestion
	const char *name;
```